packages/react-client/src/useTelnyxClient.ts

```diff
@@ -55,103 +37,38 @@ type CredentialOptions = TokenCredentialOptions | UsernameCredentialOptions;
-  // the client object through component updates
-  let telnyxClientRef = useRef<any>();
-  let [clientState, setClientState] = useState<ClientState | null>(null);
-  let [call, setCall] = useState<IPartialCall | null>(null);
-
-  const updateWebRTCState = (state: ClientState) => {
-    if (isMountedRef.current) {
-      setClientState(state);
-    }
-  };
-
-  useEffect(() => {
-    isMountedRef.current = true;
+): TelnyxRTC | undefined {
+  const telnyxClientRef = useRef<TelnyxRTC>();
 
-    if (!credentialParam) {
-      return;
-    }
+  if (process.env.NODE_ENV === 'development' && telnyxClientRef.current) {
+    console.warn(
+      'Instance of Telnyx Client already exists. Did you mean to create multiple instances of Telnyx Client?'
+    );
+  }
 
-    const telnyxClient = new TelnyxRTC({
-      ...credentialParam,
-      ...clientOptions,
-    });
+  telnyxClientRef.current = new TelnyxRTC({
+    ...credentialParam,
+    ...clientOptions,
+  });
 
-    telnyxClient.on('telnyx.ready', () => {
-      updateWebRTCState(ClientState.READY);
-    });
+  telnyxClientRef.current.on('telnyx.error', () => {
+    telnyxClientRef.current?.disconnect();
+  });
 
-    telnyxClient.on('telnyx.error', () => {
-      telnyxClient.disconnect();
-
-      updateWebRTCState(ClientState.ERROR);
-    });
-
-    telnyxClient.on('telnyx.socket.error', () => {
-      telnyxClient.disconnect();
-
-      updateWebRTCState(ClientState.ERROR);
-    });
-
-    telnyxClient.on('telnyx.socket.close', () => {
-      updateWebRTCState(ClientState.DISCONNECTED);
-    });
-
-    telnyxClient.on('telnyx.notification', (notification: any) => {
-      if (notification.call) {
-        const {
-          state,
-          direction,
-          options,
-          answer,
-          hangup,
-          muteAudio,
-          unmuteAudio,
-          remoteStream,
-        } = notification.call;
-
-        if (state === 'hangup' || state === 'destroy') {
-          setCall(null);
-        } else {
-          setCall({
-            state,
-            direction,
-            options,
-            remoteStream,
-            answer: answer.bind(notification.call),
-            hangup: hangup.bind(notification.call),
-            muteAudio: muteAudio.bind(notification.call),
-            unmuteAudio: unmuteAudio.bind(notification.call),
-          });
-        }
-      }
-    });
-
-    telnyxClientRef.current = telnyxClient;
+  telnyxClientRef.current.on('telnyx.socket.error', () => {
+    telnyxClientRef.current?.disconnect();
+  });
 
+  useEffect(() => {
     // IDEA Allow caller to defer connect
-    telnyxClientRef.current.connect();
+    telnyxClientRef.current?.connect();
 
     return () => {
-      isMountedRef.current = false;
-
       telnyxClientRef.current?.disconnect();
-      telnyxClientRef.current = undefined;
     };
-  }, [credentialParam]);
+  }, []);
 
-  return { client: telnyxClientRef.current, call, clientState };
+  return telnyxClientRef.current;
 }
```

Comment:
To make sure all this code is only running in the browser, we should run it all inside `useEffect` I believe.

Reply:
If you instantiate inside the `useEffect` you would have to return the ref from the function, not the `.current`

Reply:
@hugobdev I don't think `TelnyxRTC` needs access to the DOM any more since we no longer provider an audio or video element ID during client instantiation. Running this in a hook would also require all consumers to check for the existence of the client, which I don't think necessary. Did you have another concern in mind?